### PR TITLE
Port Github specific guides from Wiki

### PR DESCRIPTION
**Pull request by Toakan**
Pulled two Github pages from the wiki and converted them. I did drop the information from contributing and separated it into the "how to sign-off" page.

https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project

https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.

Ran through the `mkdocs serv` command to check accuracy.

image

Signed-off-by: Tom Daykin <tdaykin@live.co.uk>

### CODE REVIEW

**Comment by Toakan** [comment]
@Mcat12 changed as requested.

File: mkdocs.yml [hunk 1]
```diff
@@ -78,6 +78,9 @@ pages:
     - 'Optional: Dual operation: LAN & VPN at the same time': 'guides/vpn/dual-operation.md'
     - 'Optional: Full and DNS-only': 'guides/vpn/dual-VPN.md'
     - 'Optional: Dynamic DNS': 'guides/vpn/dynDNS.md'
+  - 'Contributing':
+          - 'Contributing on Github': guides/github/contributing.md
+          - 'How to sign-off commits': guides/github/how-to-signoff.md
```
[the remark this answers is not in the document]
Comment (Toakan) [reply mkdocs.yml]: I'm guessing you meant the indent spaces when setting a sub-menu. This has been fixed, note that my YAML parsing didn't complain about it so not sure how exact mkdocs is with interpretation.

File: mkdocs.yml [hunk 2]
```diff
@@ -78,6 +78,9 @@ pages:
     - 'Optional: Dual operation: LAN & VPN at the same time': 'guides/vpn/dual-operation.md'
     - 'Optional: Full and DNS-only': 'guides/vpn/dual-VPN.md'
     - 'Optional: Dynamic DNS': 'guides/vpn/dynDNS.md'
+  - 'Contributing':
```
[the remark this answers is not in the document]
Comment (Toakan) [reply mkdocs.yml]: Side note, should the current directory level stay the same `docs/guides/gihub` or should that be moved up to `docs/github` ?

File: docs/guides/github/contributing.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+> 2. The contribution is based upon previous work that, to the best of my knowledge, is covered under an > appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or
+
+> 3. The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.
+
+> 4. I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.
+
+Contributors *sign-off* that they adhere to these requirements by adding a Signed-off-by line to commit messages. See [this guide for how to sign-off correctly](./how-to-signoff.md).
+
+
+#### DCO Failures
```
[the remark this answers is not in the document]
Comment (Toakan) [reply docs/guides/github/contributing.md]: I've added a screenshot of the correct DCO message and mentioned the bot, however i'm bad at waffling on so if you'd like something more structured in it's place please tell me!

File: docs/guides/github/how-to-signoff.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+Pi-hole requires a sign-off message in the following format appear on each commit in the pull request:
+```
+This is my commit message
+
+Signed-off-by: Random J Developer <random@developer.example.org>
+```
+The text can either be manually added to your commit body, or you can add either -s or --signoff to your usual git commit commands.
```
[the remark this answers is not in the document]
Comment (Toakan) [reply docs/guides/github/how-to-signoff.md]: Fixed these two lines.

File: mkdocs.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mcat12) [reply mkdocs.yml]: It should be moved to `docs/github`

File: docs/guides/github/contributing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mcat12) [reply docs/guides/github/contributing.md]: The screenshot can be cropped to only include the "checks" section.

**Comment by Toakan** [comment]
@Mcat12 Cropped as requested.

**Comment by Mcat12** [comment]
It would be good to have the screenshot show the bot as well, so you can see what it looks like.

**Comment by DL6ER** [comment]
@Mcat12 Is this a hard requirement for this PR?

**Comment by Mcat12** [comment]
It would be nice, since otherwise all you see is "All checks have passed" but no information about what check to look for DCO information.

**Comment by Toakan** [comment]
@Mcat12 which part of the bot do you want to show here? For the most part the simple checks at the bottom of the PR is what people will see. 

If the PR fails the DCO, they get notification in the PR to that fact?

**Comment by Mcat12** [comment]
Include the "DCO Successful in -1m — DCO" part along with what you already have (the "All checks have passed" part).

**Comment by Toakan** [comment]
Expanded screenshot added.

File: docs/guides/github/contributing.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+The Pi-hole project uses a DCO bot for all GitHub pulls to verify that each commit is signed off. When you create your pull request, it will automaticaly be verified by this bot. An example of what to expect is below.
+
+![Screenshot](../../images/github-commit-bots.png)
+
+If your Pull Request fails the DCO check, it's necessary to fix the entire commit history in the PR. Although this is a situation we'd like to avoid the best practice is to squash the commit history to a single commit, append the DCO sign-off as described above or interactively in the rebase comment editing process, and force push. For example, if you have 2 commits in your history (Note the ~2):
+
+An example of the correct bot response is below.
+
+![Screenshot](../../images/github-commit-bots-expanded.png)
```
Comment (Mcat12) [inline docs/guides/github/contributing.md]: This image should replace the one above.

**Comment by Toakan** [comment]
Updated @Mcat12

**Pull request merged**